internal/cmd/plugin/command_install.go

```
@@ -0,0 +1,141 @@
+
+	return nil
+}
+
+func getPluginManifest(pluginName, dir string) (*Manifest, error) {
+	files, err := os.ReadDir(dir)
+	if err != nil {
+		return nil, err
+	}
+
+	for _, file := range files {
+		if file.Name() != pluginName || !file.IsDir() {
+			continue
+		}
+
+		manifestPath := fmt.Sprintf("%s/%s/manifest.yml", dir, file.Name())
+		if !utils.DoesPathExist(manifestPath) {
+			return nil, errors.Errorf("manifest not found for plugin %s", pluginName)
+		}
+
+		manifestFile, err := os.ReadFile(manifestPath)
+		if err != nil {
+			return nil, err
+		}
+
+		manifest := new(Manifest)
+		if err := yaml.Unmarshal(manifestFile, manifest); err != nil {
+			return nil, err
+		}
+		manifest.Name = file.Name()
+
+		return manifest, nil
+	}
+
+	return nil, errors.Errorf("plugin %s not found", pluginName)
+}
+
+func installPlugin(manifest *Manifest, repositoryDir, installDir string) error {
+	language, ver := getLanguage(manifest)
+
+	var pluginInstaller plugin.PluginInstaller
+	switch language {
+	case "go":
+		pluginInstaller = &plugin.GoPluginInstaller{Name: manifest.Name}
+	case "python":
+		pluginInstaller = &plugin.PythonPluginInstaller{Name: manifest.Name, RepositoryDir: repositoryDir, InstallDir: installDir}
+	case "shell":
+		pluginInstaller = &plugin.ShellPluginInstaller{Name: manifest.Name, RepositoryDir: repositoryDir, InstallDir: installDir}
+	default:
+		return errors.Errorf("installation of plugins using %s is not yet supported", language)
+	}
+
+	if err := pluginInstaller.CheckVersion(ver); err != nil {
+		return err
+	}
+	return pluginInstaller.Install()
+}
+
+func getLanguage(manifest *Manifest) (string, *version.Version) {
+	if manifest == nil || len(manifest.Dependencies) == 0 {
+		return "", nil
+	}
+
+	language := manifest.Dependencies[0]
+	language.Name = strings.ToLower(language.Name)
+	if language.Version == "" {
+		return language.Name, nil
+	}
+
+	ver, err := version.NewVersion(language.Version)
+	if err != nil {
+		return language.Name, nil
```

Do we really want to ignore this error? This could cause a plugin to fail without warning.